common/src/attestations/dispute_manager.rs

```diff
@@ -25,27 +24,44 @@ struct DisputeManager;
 pub fn dispute_manager(
     network_subgraph: &'static SubgraphClient,
     interval: Duration,
-) -> Eventual<Address> {
-    timer(interval).map_with_retry(
-        move |_| async move {
-            let response = network_subgraph
-                .query::<DisputeManager, _>(dispute_manager::Variables {})
-                .await
-                .map_err(|e| e.to_string())?;
-
-            response.map_err(|e| e.to_string()).and_then(|data| {
-                data.graph_network
-                    .map(|network| network.dispute_manager)
-                    .ok_or_else(|| "Network 1 not found in network subgraph".to_string())
-            })
-        },
-        move |err: String| {
-            warn!("Failed to query dispute manager: {}", err);
-
-            // Sleep for a bit before we retry
-            sleep(interval.div_f32(2.0))
-        },
-    )
+) -> Receiver<Option<Address>> {
+    let (tx, rx) = watch::channel(None);
+    tokio::spawn(async move {
+        let mut time_interval = time::interval(interval);
+        time_interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+        loop {
+            time_interval.tick().await;
+
+            let result = async {
+                let response = network_subgraph
+                    .query::<DisputeManager, _>(dispute_manager::Variables {})
+                    .await
+                    .map_err(|e| e.to_string())?;
+
+                response.map_err(|e| e.to_string()).and_then(|data| {
+                    data.graph_network
+                        .map(|network| network.dispute_manager)
+                        .ok_or_else(|| "Network 1 not found in network subgraph".to_string())
+                })
```

Comment:
One cool thing about not needing eventuals anymore is that you don't need to convert to string each error.

Could you remove all of these `map_err`? We can use `anyhow::Error`.